dev

void-85 opened the pull request.

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

Commit:
Очень непонятная работа, делаешь все по шагам как описано, но не компилируется, так компилируется и работает на эмуляторе/физ.устройстве

Commit:
Очень непонятная работа, делаешь все по шагам как описано, но не компилируется, так компилируется и работает на эмуляторе/физ.устройстве

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,44 @@
+
+
+public class BaseListDiffCallback<T>
+        extends    DiffUtil.ItemCallback<T>
```
Обычно в таком стиле не расставляют переносы, только при очень большой длине строки

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,44 @@
+{
+
+    public boolean theSameAs( T asWhat )
+    {
```
По код стилю java не нужно переносить открывающую фигурную скобку

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,44 @@
+
+public class BaseListDiffCallback<T>
+        extends    DiffUtil.ItemCallback<T>
+        implements ListDiffInterface    <T>
```
По заданию нам нужно создать дженерик класс, тип которого наследуется от ListDiffInterface, сам класс наследуется от DiffUtil.ItemCallback
В сумме получаем вот так:
```
public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
```

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -2,7 +2,9 @@
-public class FilterContactTypeUi {
+import ru.yandex.practicum.contacts.presentation.base.ListDiffInterface;
+
+public class FilterContactTypeUi{
```
А все типы, которые мы используем для типизации класса BaseListDiffCallback должны реализовывать интерфейс ListDiffInterface и соответствующим образом переопределять все методы

Commit:
Исправлено:

❌ Каждый из классов ContactUI, FilterContactTypeUI, SortTypeUI реализуют созданный интерфейс ListDiffInterface и метод theSameAs()

❌ Создан дженерик-класс BaseListDiffCallback с реализацией
